### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by i-like-robots**
This adds a new playlist component accessible on the video object. It accepts an array of Brightcove video IDs, exposes `prev` and `next` methods to 'scroll' between videos and will autoplay the next video when the current ends. It will also cache video data to minimise delay switching back and forth.

### CODE REVIEW

**Comment by JakeChampion** [comment]
The code looks good. I tried to run the tests but mentioned missing dependencies. Here is the diff of what I needed to add to get the tests running.

``` diff
diff --git i/package.json w/package.json
index c9e93e5..aee22d2 100644
--- i/package.json
+++ w/package.json
@@ -7,6 +7,7 @@
     "babel-loader": "^5.3.2",
     "babel-runtime": "^5.6.15",
     "bower-webpack-plugin": "^0.1.8",
+    "chai": "^3.5.0",
     "imports-loader": "^0.6.4",
     "json-loader": "^0.5.4",
     "karma": "^0.13.15",
@@ -18,6 +19,7 @@
     "karma-phantomjs-launcher": "^1.0.0",
     "karma-sinon": "^1.0.4",
     "karma-webpack": "^1.7.0",
+    "mocha": "^3.0.2",
     "origami-build-tools": "^5.0.0",
     "sinon": "^1.17.5",
     "webpack": "^1.12.2"
```

**Comment by i-like-robots** [comment]
Mocha is a peer dependency of the karma-mocha package and chai of karma-chai... are you Node 6-ing/NPM 3-ing?

**Comment by JakeChampion** [comment]
Yep, that will be why :-)

**Comment by i-like-robots** [comment]
Done!

File: src/js/playlist.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	next () {
+		this.goto(this.currentIndex + 1);
+	}
+
+	prev () {
+		this.goto(this.currentIndex - 1);
+	}
+
+	goto (index) {
```
Comment (AlbertoElias) [inline src/js/playlist.js:26]: I'd say this should be camel cased

File: test/playlist.test.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	let player;
+	const queue = ['foo', 'bar', 'baz', 'qux'];
+
+	beforeEach(() => {
+		player = createPlayer();
+	});
+
+	describe('constructor', () => {
+		it('can instantiate', () => {
+			const instance = new Subject({ player, queue: [] });
```
Comment (AlbertoElias) [inline test/playlist.test.js:26]: Could you call it Playlist so it's easier to read?

File: test/playlist.test.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			sinon.assert.calledOnce(player.update);
+		});
+
+		it('listens for the video to end to trigger the next in the queue', () => {
+			player.videoData = { id: 'bar' };
+
+			new Subject({ player, queue });
+
+			// no DOM so trigger this on the listener directly
+			player.containerEl.dispatchEvent(new CustomEvent('ended', { bubbles: false }));
```
Comment (AlbertoElias) [inline test/playlist.test.js:55]: As it's a unit test, how we've decided to test this is by spying on addEventListener and making sure that it´s being set and that the handler is the correct one

**Pull request merged**